Could not decode to UTF-8 column 'json_data' with text ....

I am getting the above error message. 

Log:
```text
 [ADMIN ] Starting user script with executable='/sist2-admin/scripts/test/run.sh', index_path='/sist2-admin/scan-TEST_v2-2023-12-11 15:27:05.254099.sist2', extra_args=''
 [INFO ] Instantiating the Index...
 [INFO ] Iterating through the documents...
 [INFO ] Could not decode to UTF-8 column 'json_data' with text '{"extension":"pdf","name":"DOC NAME","path":"PATH/TO/DOC
 [INFO ] [ERROR] Something went wrong with the doc loop!
 [INFO ] Finished Processing 5040 documents.
```


User Script:
```python from sist2 import Sist2Index, Sist2Document
import sys

print("Instantiating the Index...")
index = Sist2Index(sys.argv[1])

print("Iterating through the documents...")
docs = 0

try:
    for doc in index.document_iter():
        docs += 1

except Exception as error:
    print(error)
    print("[ERROR] Something went wrong with the doc loop!")

print("Finished Processing %d documents." % docs)
```

I am not sure where the non-UTF-8 data is coming from. The document identified is a PDF and it does include pages that were OCR'd during the `scan` so, maybe it came in through that?

I am unsure how to identify the character that is causing the issue. The sqlite reader I have used seems to handle it gracefully.

--- a/setup.py
+++ b/setup.py
@@ -2,7 +2,7 @@ from setuptools import setup
 
 setup(
     name="sist2-python",
-    version="2.1",
+    version="2.1.1",
     description="Set of python tools to interface with sist2 index files. Used in user scripts",
     author="simon987",
     author_email="me@simon987.net",

--- a/setup.py
+++ b/setup.py
@@ -2,7 +2,7 @@ from setuptools import setup
 
 setup(
     name="sist2-python",
-    version="2.1.1",
+    version="2.1.2",
     description="Set of python tools to interface with sist2 index files. Used in user scripts",
     author="simon987",
     author_email="me@simon987.net",

--- a/setup.py
+++ b/setup.py
@@ -2,7 +2,7 @@ from setuptools import setup
 
 setup(
     name="sist2-python",
-    version="2.0",
+    version="2.1",
     description="Set of python tools to interface with sist2 index files. Used in user scripts",
     author="simon987",
     author_email="me@simon987.net",

--- a/setup.py
+++ b/setup.py
@@ -2,7 +2,7 @@ from setuptools import setup
 
 setup(
     name="sist2-python",
-    version="2.1.2",
+    version="2.1",
     description="Set of python tools to interface with sist2 index files. Used in user scripts",
     author="simon987",
     author_email="me@simon987.net",

--- a/sist2/__init__.py
+++ b/sist2/__init__.py
@@ -46,6 +46,7 @@ class Sist2Index:
         """
         self.filename = filename
         self.conn = sqlite3.connect(filename)
+        print("hello")
         self.conn.text_factory = str
 
         self.cur = self.conn.cursor()

--- a/sist2/__init__.py
+++ b/sist2/__init__.py
@@ -46,6 +46,7 @@ class Sist2Index:
         """
         self.filename = filename
         self.conn = sqlite3.connect(filename)
+        self.conn.text_factory = str
 
         self.cur = self.conn.cursor()
 

--- a/sist2/__init__.py
+++ b/sist2/__init__.py
@@ -46,8 +46,7 @@ class Sist2Index:
         """
         self.filename = filename
         self.conn = sqlite3.connect(filename)
-        print("hello")
-        self.conn.text_factory = str
+        self.conn.text_factory = lambda b: b.decode(errors = 'ignore')
 
         self.cur = self.conn.cursor()
 

